lib/promscrape/discovery/docker/containers.go

```diff
@@ -0,0 +1,128 @@
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discoveryutils"
+)
+
+// See https://github.com/moby/moby/blob/314759dc2f4745925d8dec6d15acc7761c6e5c92/docs/api/v1.41.yaml#L4024
+type container struct {
+	Id     string
+	Names  []string
+	Labels map[string]string
+	Ports  []struct {
+		IP          string
+		PrivatePort int
+		PublicPort  int
+		Type        string
+	}
+	HostConfig struct {
+		NetworkMode string
+	}
+	NetworkSettings struct {
+		Networks map[string]struct {
+			IPAddress string
+			NetworkID string
+		}
+	}
+}
+
+func getContainersLabels(cfg *apiConfig) ([]map[string]string, error) {
+	networkLabels, err := getNetworksLabels(cfg, "__meta_docker_")
+
+	if err != nil {
+		return nil, err
+	}
+
+	containers, err := getContainers(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return addContainersLabels(containers, networkLabels, cfg.port), nil
+}
+
+func getContainers(cfg *apiConfig) ([]container, error) {
+	resp, err := cfg.getAPIResponse("/containers/json")
+	if err != nil {
+		return nil, fmt.Errorf("cannot query dockerd api for containers: %w", err)
+	}
+	return parseContainers(resp)
+}
+
+func parseContainers(data []byte) ([]container, error) {
+	var containers []container
+	if err := json.Unmarshal(data, &containers); err != nil {
+		return nil, fmt.Errorf("cannot parse containers: %w", err)
+	}
+	return containers, nil
+}
+
+func addContainersLabels(containers []container, networkLabels map[string]map[string]string, defaultPort int) []map[string]string {
+	var ms []map[string]string
+	for _, c := range containers {
+		commonLabels := map[string]string{
```

Comment:
I suggest to check for c.Names len, probably its not initialized. 

```go
		if len(c.Names) == 0 {
			continue
		}
```

Reply:
Added